proxy/pkg/config/config.go

```diff
@@ -16,11 +16,12 @@ type Config struct {
 
 	// Global bucket
 
-	PrimaryCluster          string `default:"ORIGIN" split_words:"true"`
-	ReadMode                string `default:"PRIMARY_ONLY" split_words:"true"`
-	ReplaceCqlFunctions     bool   `default:"false" split_words:"true"`
-	AsyncHandshakeTimeoutMs int    `default:"4000" split_words:"true"`
-	LogLevel                string `default:"INFO" split_words:"true"`
+	PrimaryCluster                string `default:"ORIGIN" split_words:"true"`
+	ReadMode                      string `default:"PRIMARY_ONLY" split_words:"true"`
+	ReplaceCqlFunctions           bool   `default:"false" split_words:"true"`
+	AsyncHandshakeTimeoutMs       int    `default:"4000" split_words:"true"`
+	LogLevel                      string `default:"INFO" split_words:"true"`
+	ControlConnMaxProtocolVersion uint   `default:"3" split_words:"true"`
```

Comment:
I'd probably put this as a string because what if someone wants to set `Dsev1` or `Dsev2`? They would have to provide the actual protocol code for it which users don't know. And this config setting should then be validated and startup should be aborted if the provided protocol version string can't be converted to an actual protocol version (and we should disallow anything below v2 and anything higher than v4 except dsev1 and dsev2)